src/main/groovy/net/wasdev/wlp/gradle/plugins/tasks/InstallLibertyTask.groovy

```diff
@@ -134,4 +135,38 @@ class InstallLibertyTask extends AbstractTask {
             }
         }
     }
+    
+    
+    protected boolean isInstalled() {
+        boolean isUpToDate = false;
+
+        File f = new File(project.buildDir, 'liberty-plugin-config.xml')
+        if (f == null || !f.exists()) {
+            return isUpToDate;
+        }
+
+        XmlParser pluginXmlParser = new XmlParser()
+        Node libertyPluginConfig = pluginXmlParser.parse(f)
+
+        if(project.configurations.libertyRuntime != null && !project.configurations.libertyRuntime.dependencies.isEmpty()) {
+            if (!libertyPluginConfig.getAt('assemblyArtifact').isEmpty()) {
+                project.configurations.libertyRuntime.dependencies.each { libertyArtifact ->
+                    if(libertyPluginConfig.getAt('assemblyArtifact').getAt('artifactId').text().equals(libertyArtifact.name)
+                        && libertyPluginConfig.getAt('assemblyArtifact').getAt('version').text().equals(libertyArtifact.version)
+                        && libertyPluginConfig.getAt('assemblyArtifact').getAt('groupId').text().equals(libertyArtifact.group)) {
+                            isUpToDate = true;
+                    }
+                }
+            }
+        }
+        else if (project.liberty.install.runtimeUrl != null) {
+            if (!libertyPluginConfig.getAt('runtimeUrl').isEmpty()) {
+                if( project.liberty.install.runtimeUrl.equals(libertyPluginConfig.getAt('runtimeUrl').text())) {
+                    isUpToDate = true;
```

Comment:
The conditional statements on lines 162, 163 and 164 could all be combined into one.